### PR TITLE
add new option --omit-null-hardware-buffers

### PR DESCRIPTION
**Pull request by bradgrantham-lunarg**
Add the option string, description of option, and detection of option.

Override vkGetAndroidHardwareBufferPropertiesANDROID and implement the
option.

### CODE REVIEW

**Review by locke-lunarg — APPROVED** [review]

File: framework/generated/generated_vulkan_replay_consumer.cpp [hunk 1]
```diff
@@ -5760,11 +5760,11 @@ void VulkanReplayConsumer::Process_vkGetAndroidHardwareBufferPropertiesANDROID(
...
+    auto in_device = GetObjectInfoTable().GetDeviceInfo(device);
     const struct AHardwareBuffer* in_buffer = static_cast<const struct AHardwareBuffer*>(PreProcessExternalObject(buffer, format::ApiCallId::ApiCall_vkGetAndroidHardwareBufferPropertiesANDROID, "vkGetAndroidHardwareBufferPropertiesANDROID"));
-    VkAndroidHardwareBufferPropertiesANDROID* out_pProperties = pProperties->IsNull() ? nullptr : pProperties->AllocateOutputData(1, { VK_STRUCTURE_TYPE_ANDROID_HARDWARE_BUFFER_PROPERTIES_ANDROID, nullptr });
+    pProperties->IsNull() ? nullptr : pProperties->AllocateOutputData(1, { VK_STRUCTURE_TYPE_ANDROID_HARDWARE_BUFFER_PROPERTIES_ANDROID, nullptr });
 
-    VkResult replay_result = GetDeviceTable(in_device)->GetAndroidHardwareBufferPropertiesANDROID(in_device, in_buffer, out_pProperties);
+    VkResult replay_result = OverrideGetAndroidHardwareBufferPropertiesANDROID(GetDeviceTable(in_device->handle)->GetAndroidHardwareBufferPropertiesANDROID, returnValue, in_device, in_buffer, pProperties);
     CheckResult("vkGetAndroidHardwareBufferPropertiesANDROID", returnValue, replay_result);
```
Comment (andrew-lunarg) [inline framework/generated/generated_vulkan_replay_consumer.cpp:5768]: Trivia: some of these lines are well over the 132 width limit from the guide (IIRC). Is some hook going to clang-format them?

**Review by andrew-lunarg — APPROVED** [review]
A couple of points that that may be trivial and another that is more of a question.

File: framework/generated/generated_vulkan_replay_consumer.cpp (hunk 1 above)
Comment (bradgrantham-lunarg) [reply framework/generated/generated_vulkan_replay_consumer.cpp:5768]: This file is generated, and we don't apply the style guide to generated files.  ([`framework/generated/.clang-format`](https://github.com/LunarG/gfxreconstruct/blob/dev/framework/generated/.clang-format) specifies `DisableFormat`)  You could make an argument that we should, I think?  Maybe that should be a task - to run the generated output through clang-format.  @andrew-lunarg  Would you like to add that task as an issue?

File: tools/replay/replay_settings.h [hunk 2]
```diff
@@ -53,6 +53,7 @@ static void PrintUsage(const char* exe_name)
     GFXRECON_WRITE_CONSOLE("\t\t\t[--opcd | --omit-pipeline-cache-data] [--wsi <platform>]");
     GFXRECON_WRITE_CONSOLE("\t\t\t[--dcp | --discard-cached-psos] [--surface-index <N>]");
     GFXRECON_WRITE_CONSOLE("\t\t\t[--remove-unsupported] [--validate]");
+    GFXRECON_WRITE_CONSOLE("\t\t\t[--omit-null-hardware-buffers]");
```
[the remark this answers is not in the document]
Comment (bradgrantham-lunarg) [reply tools/replay/replay_settings.h:56]: Good catch!

File: tools/tool_settings.h [hunk 3]
```diff
@@ -76,6 +76,7 @@ const char kRemoveUnsupportedOption[]          = "--remove-unsupported";
 const char kValidateOption[]                   = "--validate";
 const char kDebugDeviceLostOption[]            = "--debug-device-lost";
 const char kCreateDummyAllocationsOption[]     = "--create-dummy-allocations";
+const char kOmitNullHardwareBuffersOption[]    = "--omit-null-hardware-buffers";
```
[the remark this answers is not in the document]
Comment (bradgrantham-lunarg) [reply tools/tool_settings.h:79]: Also good catch!

[the remark this answers is not in the document]
Comment (bradgrantham-lunarg) [reply tools/tool_settings.h]: Fixed

File: tools/replay/replay_settings.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (bradgrantham-lunarg) [reply tools/replay/replay_settings.h]: fixed

**Review by panos-lunarg — APPROVED** [review]
lgtm

**Review by locke-lunarg — APPROVED** [review]

File: framework/generated/generated_vulkan_replay_consumer.cpp (hunk 1 above)
Comment (andrew-lunarg) [reply framework/generated/generated_vulkan_replay_consumer.cpp:5768]: I can add that. My take: the formatting would be done in the generator, but still disabled in the `framework/generated/.clang-format` so editors and other tools know to stay out of there.

Comment (bradgrantham-lunarg) [reply framework/generated/generated_vulkan_replay_consumer.cpp:5768]: That's a good idea and sounds right.  Thanks!

**Review by andrew-lunarg — APPROVED** [review]

**Pull request merged**